bokeh/application/handlers/code_runner.py

```diff
@@ -78,3 +86,8 @@ def run(self, module, post_check):
             filename, line_number, func, txt = traceback.extract_tb(exc_traceback)[-1]
 
             self._error = "%s\nFile \"%s\", line %d, in %s:\n%s" % (str(e), os.path.basename(filename), line_number, func, txt)
+
+        finally:
+            # undo sys.path, CWD fixups
+            del sys.path[0]
```

Comment:
Should assert that `sys.path[0]` is what you inserted...?  Or put in a check that sensibly handles the case when it's not.